ably-ios/ARTRealtimeChannel.m

```diff
@@ -148,6 +148,28 @@ - (void)publishProtocolMessage:(ARTProtocolMessage *)pm cb:(ARTStatusCallback)cb
     }
 }
 
+
+- (void)sendMessage:(ARTProtocolMessage *)pm cb:(ARTStatusCallback)cb {
+    __block BOOL gotFailure = false;
+    NSString *oldConnectionId = self.realtime.connection.id;
+    __block ARTEventListener *listener = [self.realtime.connection on:^(ARTConnectionStateChange *stateChange) {
+        if (!(stateChange.current == ARTRealtimeClosed || stateChange.current == ARTRealtimeFailed
+              || (stateChange.current == ARTRealtimeConnected && ![oldConnectionId isEqual:self.realtime.connection.id] /* connection state lost */))) {
+            return;
```

Comment:
I had to revert this at 2ed75ea. The listener only has to be unsubscribed if there's a failure or the message is sent. The connection can change several times before the message callback arrives, e. g. when reconnecting.

RealtimeClientConnection and RealtimeClientChannel tests are passing now on my machine, let's see if the whole suite does... (I'm not running all of them in my machine because that basically blocks me from working for half an hour.)